test/test_arbitrage.py

```diff
@@ -104,22 +110,184 @@ def setUp(self):
         self.market_1.add_listener(MarketEvent.OrderFilled, self.market_1_order_fill_logger)
         self.market_2.add_listener(MarketEvent.OrderFilled, self.market_2_order_fill_logger)
 
-    def test_find_profitable_arbitrage_orders(self):
-        pass
+    def test_ready_for_new_orders(self):
+        # No pending orders
+        self.assertTrue(self.strategy.ready_for_new_orders(self.market_pair))
+
+        self.clock.backtest_til(self.start_timestamp + 6)
+        # prevent making new orders
+        self.market_1.set_balance("COINALPHA", 0)
+
+        self.assertFalse(self.strategy.ready_for_new_orders(self.market_pair))
+
+        # run till market orders complete and cool off period passes
+        self.clock.backtest_til(self.start_timestamp + 20)
+
+        self.assertTrue(self.strategy.ready_for_new_orders(self.market_pair))
+
+    def test_arbitrage_profitable(self):
+        self.market_1.set_balance("COINALPHA", 5)
+        self.market_2.set_balance("COINALPHA", 5)
+        self.clock.backtest_til(self.start_timestamp + 1)
+        market_orders = self.strategy.tracked_taker_orders
+        market_1_market_order = [order for market, order in self.strategy.tracked_taker_orders
+                                 if market == self.market_1][0]
+        market_2_market_order = [order for market, order in self.strategy.tracked_taker_orders
+                                 if market == self.market_2][0]
+
+        self.assertTrue(len(market_orders) == 2)
+        self.assertEqual(5, market_1_market_order.amount)
+        self.assertEqual(self.start_timestamp + 1, market_1_market_order.timestamp)
+        self.assertEqual(5, market_2_market_order.amount)
+        self.assertEqual(self.start_timestamp + 1, market_2_market_order.timestamp)
+
+    def test_arbitrage_not_profitable(self):
+
+        self.market_2_data.order_book.apply_diffs(
+            [OrderBookRow(1.05, 1.0, 2)],
+            [], 2)
+        self.clock.backtest_til(self.start_timestamp + 1)
+        market_orders = self.strategy.tracked_taker_orders
+        self.assertTrue(len(market_orders) == 0)
 
     def test_find_best_profitable_amount(self):
-        pass
+        self.market_2_data.order_book.apply_diffs(
+            [OrderBookRow(1.1, 30, 2)],
+            [],
+            2
+        )
+        """
+           raw_profitability  bid_price_adjusted  ask_price_adjusted  bid_price  ask_price  step_amount
+        0           1.039801               1.045               1.005        1.1      1.005         10.0
+        1           1.029557               1.045               1.015        1.1      1.015         20.0
+        """
+        amount, profitability = self.strategy.find_best_profitable_amount(*self.market_pair)
+        self.assertEqual(30.0, amount)
+        self.assertEqual(1.0329489291598024, profitability)
```

Comment:
If you're checking for equality of floating point numbers, always use `assertAlmostEqual()`.

It can also help you use a shorter decimal representation as the model answer.